pkg/fleet-manager/pipeline/render/trigger.go

```diff
@@ -0,0 +1,163 @@
+/*
+Copyright Kurator Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package render
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	pipelineapi "kurator.dev/kurator/pkg/apis/pipeline/v1alpha1"
+)
+
+const (
+	TriggerTemplateName = "pipeline trigger template"
+)
+
+type TriggerConfig struct {
+	PipelineName      string
+	PipelineNamespace string
+	OwnerReference    *metav1.OwnerReference
+	AccessMode        string
+	StorageRequest    string
+	StorageClassName  string
+	VolumeMode        string
+}
+
+// ServiceAccountName is the service account used by trigger
+func (cfg TriggerConfig) ServiceAccountName() string {
+	return cfg.PipelineName
+}
+
+// RenderTriggerWithPipeline takes a pipeline object and generates YAML byte array configuration representing the trigger configuration.
+func RenderTriggerWithPipeline(pipeline *pipelineapi.Pipeline) ([]byte, error) {
+	config := TriggerConfig{
+		PipelineName:      pipeline.Name,
+		PipelineNamespace: pipeline.Namespace,
+		OwnerReference:    GeneratePipelineOwnerRef(pipeline),
+	}
+	if pipeline.Spec.SharedWorkspace != nil {
+		config.AccessMode = string(pipeline.Spec.SharedWorkspace.AccessMode)
+		config.StorageRequest = pipeline.Spec.SharedWorkspace.StorageRequest
+		config.StorageClassName = pipeline.Spec.SharedWorkspace.StorageClassName
+		config.VolumeMode = string(pipeline.Spec.SharedWorkspace.VolumeMode)
+	}
+
+	return RenderTrigger(config)
+}
+
+// RenderTrigger takes a TriggerConfig object and generates YAML byte array configuration representing the trigger configuration.
+func RenderTrigger(cfg TriggerConfig) ([]byte, error) {
+	return renderTemplate(TriggerTemplateContent, TriggerTemplateName, cfg)
```

Comment:
same as other comments. do we need white space for TriggerTemplateName, as i cannot see the details

Reply:
Actually, 'TemplateName' is not a functioning variable, and it can contain spaces in the middle. see https://github.com/kurator-dev/kurator/blob/09a35ebf3b8960ab18497c56449bb53404c4ca2d/pkg/fleet-manager/plugin/render.go#L93

IMHO, It is only required as a parameter for creating Go templates and more like a key for tpls.

Reply:
of cause, we can use "-" instead of space

Reply:
IC, it is good